modules/slugs/includes/qtranslate-slug-settings.php

```diff
@@ -310,37 +186,44 @@ function qts_sanitize_bases( $base_slugs ) {
  */
 function qts_show_settings_page() {
     $settings_output = qts_get_settings();
+
+    if ( empty( $settings_output['qts_page_sections'] ) ) {
+        return;
+    }
+    QTX_Admin_Settings::open_section( 'slugs' );
     ?>
-    <div class="wrap">
-        <div class="icon32" id="icon-options-general"></div>
-        <h2><?php echo $settings_output['qts_page_title']; ?></h2>
-        <p><?php _e( 'If you activated previously the <a href="options-permalink.php">pretty permalinks</a>, in this section you can translate the <abbr title="en inglés, Universal Resource Locator">URLs</abbr> <strong>bases</strong> for <a href="https://codex.wordpress.org/Function_Reference/register_post_type#Arguments">public</a> post_types, categories, tags and taxonomies.', 'qts' ); ?> </p>
-
-        <form action="options.php" method="post">
-            <?php
-            // https://codex.wordpress.org/Function_Reference/settings_fields
-            settings_fields( $settings_output['qts_option_name'] );
-
-            // https://codex.wordpress.org/Function_Reference/do_settings_sections
-            do_settings_sections( __FILE__ );
-            // rewrite rules
-            flush_rewrite_rules();
-            ?>
-            <p class="submit">
-                <input name="Submit" type="submit" class="button-primary"
-                       value="<?php esc_attr_e( 'Save Changes', 'qts' ); ?>"/>
-            </p>
-
-        </form>
-    </div><!-- wrap -->
-<?php }
+     <p class="heading"><?php _e( 'If you activated previously the <a href="options-permalink.php">pretty permalinks</a>, in this section you can translate the <abbr title="en inglés, Universal Resource Locator">URLs</abbr> <strong>bases</strong> for <a href="https://developer.wordpress.org/reference/functions/register_post_type/#parameters">public</a> post_types, categories, tags and taxonomies.', 'qts' ); ?> </p>
+
+    <?php foreach ( $settings_output['qts_page_sections'] as $id => $title ) { ?>
+        <h2><?php echo $title; ?></h2>
+        <?php qts_section_fn($id);
+        // call the "add_settings_field" for each!
+        foreach ( $settings_output['qts_page_fields'] as $option ) {
+            if ( $option['section']==$id ) {
+                qts_show_form_field( $option );
+            }
+        }
+    }
+    QTX_Admin_Settings::close_section( 'slugs' );
+    flush_rewrite_rules();
```

Comment:
I see this `flush_rewrite_rules` comes from the old form. Do we need this when showing the options? Shouldn't it rather be done on save?

Reply:
agree, ref 578e24f633225663823619506a2840d1ae042716

Reply:
Great!